pkg/apis/toolchain/v1alpha1/hostoperatorconfig_types.go

```diff
@@ -0,0 +1,89 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HostOperatorConfigSpec contains all configuration parameters of the host operator
+// +k8s:openapi-gen=true
+type HostOperatorConfigSpec struct {
+	// Keeps parameters necessary for automatic approval
+	// +optional
+	AutomaticApproval AutomaticApproval `json:"automaticApproval,omitempty"`
+}
+
+// Defines all parameters necessary for automatic approval
+// +k8s:openapi-gen=true
+type AutomaticApproval struct {
+	// Defines if the automatic approval is activated or not
+	Activated bool `json:"activated"`
+
+	// Contains threshold (in percentage of usage) that defines when the automatic approval should be stopped
+	// +optional
+	ResourceCapacityThreshold ResourceCapacityThreshold `json:"resourceCapacityThreshold,omitempty"`
+
+	// Defines what is the maximal number of users to be allowed for automatic approval.
+	// When the number is reached, then the automatic approval is stopped.
+	// +optional
+	MaxNumberOfUsers MaxNumberOfUsers `json:"maxNumberOfUsers,omitempty"`
+}
+
+// Contains default capacity threshold as well as specific ones for particular member clusters
+// +k8s:openapi-gen=true
+type ResourceCapacityThreshold struct {
+	// It is the default capacity threshold (in percentage of usage) to be used for all member clusters if no special threshold is defined
+	DefaultThreshold int `json:"defaultThreshold"`
+
+	// Contains a map of specific capacity thresholds (in percentage of usage) for particular member clusters mapped by their names
+	// +optional
+	// +patchStrategy=merge
```

Comment:
Shouldn't we add `// +mapType=atomic`?
```suggestion
	// +patchStrategy=merge
	// +mapType=atomic
```

Reply:
Thanks @alexeykazakov - I totally forgot about this annotation. :man_facepalming: 
I guess that I can remove the `patchStrategy` since it doesn't make any sense to have it there.